api/server/router/image/image_routes.go

```diff
@@ -64,11 +56,7 @@ func (s *router) postCommit(ctx context.Context, w http.ResponseWriter, r *http.
 		Config:  c,
 	}
 
-	if !s.daemon.Exists(cname) {
-		return derr.ErrorCodeNoSuchContainer.WithArgs(cname)
-	}
-
-	imgID, err := dockerfile.Commit(cname, s.daemon, commitCfg)
+	imgID, err := s.backend.ImageCommit(cname, commitCfg)
```

Comment:
FYI, as part of removing builder's code dependency on daemon, we were thinking of using the new API client call such as ContainerCommit on the server side as well to make a call to the unix socket on local host.  This will remove any code-level daemon dependency. Eventually, when the builder's parser/evaluator moves client-side, then the server-side would be much more simpler.